### PR TITLE
Start preparation for TIM2025 workshop

### PR DESCRIPTION
**Pull request by anwai98**
WIP!

### CODE REVIEW

File: workshops/tim_2025/README.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+# TIM 2025 Workshop: Segment Anything for Microscopy
+
+This document walks you through the preparation for the upcoming TIM 2025 workshop on "Segment Anything for Microscopy".
+
+## Workshop Overview
+
+The workshop will be three hours and will be divided into three parts:
+1. Step 0: Install `micro-sam` on HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).
+2. Short introduction (30 minutes, you can find the slides [here](TODO))
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:9]: 30 minutes is too long for this.

File: workshops/tim_2025/README.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+3. Using the `micro_sam` napari plugin (on HIVE) for interactive (and automatic) 2d and 3d segmentation (30 minutes).
+4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model or an advanced application (60 minutes).
+5. Discussion on "What you can do next" (30 minutes)
+
+We will walk through how to use the `micro_sam` plugin for interactive segmentation in part 3, so that you can then apply it to your own data (or the example data that is most similar to your targeted application) in part 4.
+Alternatively, you can also work on model finetuning or an advanced application, such as using the `micro_sam` in part 4.
+And finally, we have an open discussion on your experience and expectations from `micro_sam` in part 5, and the take home segmentation workflow for you all.
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:16]:
```suggestion
And finally, we have a discussion on your experience and feedback / possible improvements in part 5. We will also try to formulate a segmentation workflow applicable for your data "at home".
```

File: workshops/tim_2025/README.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ This document walks you through the preparation for the upcoming TIM 2025 worksh
 
 The workshop will be three hours and will be divided into three parts:
 1. Step 0: Install `micro-sam` on HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:8]:
```suggestion
1. Step 0: Install `micro-sam` on your laptop or HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).
```

File: workshops/tim_2025/README.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ This document walks you through the preparation for the upcoming TIM 2025 worksh
...
-2. Short introduction (30 minutes, you can find the slides [here](TODO))
+2. Short introduction (15 minutes, you can find the slides [here](https://docs.google.com/presentation/d/1jEmjHLOCMefzU8W3hBMGh7ooseEIfoe-/edit?usp=sharing&ouid=113044948772353505255&rtpof=true&sd=true))
 3. Using the `micro_sam` napari plugin (on HIVE) for interactive (and automatic) 2d and 3d segmentation (30 minutes).
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:10]:
```suggestion
3. Using the `micro_sam` napari plugin for interactive (and automatic) 2d and 3d segmentation (30 minutes).
```

File: workshops/tim_2025/README.md [hunk 5]
```diff
@@ -6,7 +6,7 @@ This document walks you through the preparation for the upcoming TIM 2025 worksh
...
 4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model or an advanced application (60 minutes).
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:11]:
```suggestion
4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model , or following an advanced example application (60 minutes).
```

File: workshops/tim_2025/README.md [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+
+The workshop will be three hours and will be divided into three parts:
+1. Step 0: Install `micro-sam` on HIVE and discuss instructions for installing `micro-sam` in your facility (30 minutes).
+2. Short introduction (15 minutes, you can find the slides [here](https://docs.google.com/presentation/d/1jEmjHLOCMefzU8W3hBMGh7ooseEIfoe-/edit?usp=sharing&ouid=113044948772353505255&rtpof=true&sd=true))
+3. Using the `micro_sam` napari plugin (on HIVE) for interactive (and automatic) 2d and 3d segmentation (30 minutes).
+4. Using the plugin on your own or on example data or data imaged in another TIM workshop, finetuning a custom model or an advanced application (60 minutes).
+5. Discussion on "What you can do next" (30 minutes)
+
+We will walk through how to use the `micro_sam` plugin for interactive segmentation in part 3, so that you can then apply it to your own data (or the example data that is most similar to your targeted application) in part 4.
+Alternatively, you can also work on model finetuning or an advanced application, such as using the `micro_sam` in part 4.
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:15]:
```suggestion
Alternatively, you can also work on model finetuning or an advanced application, such as using the `micro_sam` pythhon library in part 4.
```

File: workshops/tim_2025/README.md [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+And finally, we have a discussion on your experience and feedback / possible improvements in part 5. We will also try to formulate a segmentation workflow applicable for your data "at home".
+
+**Please read the [Workshop Preparation](#workshop-preparation) section carefully and follow the relevant steps to get started.**
+
+## Workshop Preparation
+
+To get started for the workshop, please do the following:
+- Install the latest version of `micro_sam`, see [Installation](#installation) for details
+- Download the models and pre-computed embeddings for the common 3D segmentation, see [here](#download-embeddings-for-3d-segmentation).
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:24]:
```suggestion
- Download the models and pre-computed embeddings for the 3D segmentation problems, see [here](#download-embeddings-for-3d-segmentation).
```

File: workshops/tim_2025/README.md [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+### Installation
+
+Please make sure to install the latest version of `micro_sam` before the workshop using `conda` (or `mamba`).
+You can create a new environment and install it like this:
+
+```bash
+conda create -n micro_sam -c conda-forge python=3.11 natsort
+conda activate micro_sam
+conda install -c conda-forge "micro_sam>=1.3.1" "pytorch>=2.5" "protobuf<5" "nifty=1.2.1=*_4"
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:43]: We need to update these instructions so that it matches what I announced in the mattermost. I will follow up on this.

File: workshops/tim_2025/README.md [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+        <li>Run `E:\TiM2025-Software\Miniforge3-Windows-x86_64.exe` in your termnial to install MiniForge. It will guide you through all instructions. Please follow along to install MiniForge successfully.</li>
+        <li>Once the installation is done, you need to initialize your conda environments using `.\miniforge3\Scripts\conda.exe init powershell`.</li>
+        <li>And here onwards, follow the installation instructions above.</li>
+    </ol>
+</details>
+
+
+### Download Embeddings for 3D EM Segmentation
+
+We provide a script to download the models used in the workshop. To run the script you first need to use `git` to download this repository:
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:62]:
```suggestion
We provide a script to download the models used in the workshop. To run the script you first need to download this repository. If you haven't done this already, then please use `git` to download it:
```

File: workshops/tim_2025/README.md [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+
+```bash
+git clone https://github.com/computational-cell-analytics/micro-sam
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:65]: Need to checkout dev

File: workshops/tim_2025/README.md [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+```
+
+This will download the data to the folder `data/cells` with images stored in the subfolder `images` and segmentation masks in `masks`. After this you can start the image series annotation tool, either via the napari plugin (we will show this in the workshop) or via the command line:
+
+```bash
+micro_sam.image_series_annotator -i data/cells/images -o annotations/cells -e embeddings/cells/vit_b_lm -m vit_b_lm
+```
+
+Note: You can use `micro_sam` with different models: the original models from Segment Anything and models finetuned for different microscopy segmentation tasks by us.
+For cell segmentation you can either use `vit_b` (the original model) or `vit_b_lm` (our model). Our `vit_b_lm` model will be better for most cell segmentation tasks but there may be cases where `vit_b` is better, so it makes sense to test both before annotating your data. Please refer to [our documentation](https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#finetuned-models) for details on the models.
```
Comment (constantinpape) [inline workshops/tim_2025/README.md:106]:
```suggestion
For cell segmentation you can either use `vit_b` (the original SAM) or `vit_b_lm` (our model; the default model used by `micro_sam`). Our `vit_b_lm` model will be better for most cell segmentation tasks but there may be cases where `vit_b` is better, so it makes sense to test both before annotating your data. Please refer to [our documentation](https://computational-cell-analytics.github.io/micro-sam/micro_sam.html#finetuned-models) for details on the models.
```

**Review by constantinpape — COMMENTED** [review]
A few more updates to the installation instructions needed (I will take care of this).
The rest looks good!

File: workshops/tim_2025/README.md (hunk 8 above)
Comment (anwai98) [reply workshops/tim_2025/README.md:43]: Okie.

File: workshops/tim_2025/README.md (hunk 10 above)
Comment (anwai98) [reply workshops/tim_2025/README.md:65]: Ahha yes you are right. I am testing this on HIVE now, and will add once your review is done.

**Comment by constantinpape** [comment]
I will go ahead and merge it, the MAC tests are still timing out because the respective fix from dev is not yet merged.
